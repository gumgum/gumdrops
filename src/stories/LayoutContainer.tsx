import React, { ReactChild, ReactChildren } from 'react';
import cx from 'classnames';

const baseClass = 'gds-layout__container';

export interface LayoutContainerProps {
    children: ReactChildren | ReactChild;
    className?: string;
    fullWidth?: boolean;
}

export const LayoutContainer: React.FC<LayoutContainerProps> = ({
    className,
    fullWidth,
    children,
    ...props
}) => {
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--full-width`]: fullWidth
    });

    return (
        <div className={rootClass} {...props}>
            {children}
        </div>
    );
};
