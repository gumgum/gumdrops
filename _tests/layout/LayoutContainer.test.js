import renderer from 'react-test-renderer';
import React from 'react';
import LayoutContainer from '../../components/layout/LayoutContainer';

test('Expect <LayoutContainer> to render properly', () => {
    const tree = renderer
        .create(
            <LayoutContainer>
                <div>Content</div>
            </LayoutContainer>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <LayoutContainer> to render as full width', () => {
    const props = {
        fullWidth: true
    };
    const tree = renderer
        .create(
            <LayoutContainer {...props}>
                <div>Content</div>
            </LayoutContainer>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
