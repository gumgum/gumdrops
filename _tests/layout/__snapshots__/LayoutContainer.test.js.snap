// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <LayoutContainer> to render as full width 1`] = `
<div
  className="gds-layout__container gds-layout__container--full-width"
>
  <div>
    Content
  </div>
</div>
`;

exports[`Expect <LayoutContainer> to render properly 1`] = `
<div
  className="gds-layout__container"
>
  <div>
    Content
  </div>
</div>
`;
